sql/query_flags.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sql
+
+// Most QFlags indicate the presence of a specific AST node.
+// Flags with more complex meanings have more specific comments.
+const (
+	QFlagNull int = iota
+
+	QFlagShowWarnings
+	QFlagInsert
+	QFlagUpdate
+	QFlagDelete
+	QFlagScalarSubquery
+	QFlagRelSubquery
+	QFlgNotExpr
+	QFlagCount
+	QFlagCountStar
+	QFlagAlterTable
+	QFlagCrossJoin
+	QFlagSort
+	QFlagFilter
+	QFlagAggregation
+	QFlagSetDatabase
+	QFlagStar
+	QFlagInnerJoin
+	QFlagLimit
+	QFlagInterval
+
+	// QFlagMax1Row indicates that a query can only return at most one row
+	QFlagMax1Row
+)
+
+type QueryFlags struct {
+	Flags FastIntSet
+}
+
+func (qp *QueryFlags) Set(flag int) {
+	if qp == nil {
+		return
+	}
+	qp.Flags.Add(flag)
+}
+
+func (qp *QueryFlags) IsSet(flag int) bool {
```

Comment:
"Everything is set if it's nil" is kind of insane bro

If you want this kind of default behavior, put it in a function up one abstraction layer (in the analyzer package), let this be a dumb data object.